src/index.cpp

```diff
@@ -2671,8 +2498,8 @@ template <typename T, typename TagT, typename LabelT> void Index<T, TagT, LabelT
         if ((new_location[old] < _max_points) // If point continues to exist
             || (old >= _max_points && old < _max_points + _num_frozen_pts))
         {
-            new_adj_list.reserve(_final_graph[old].size());
-            for (auto ngh_iter : _final_graph[old])
+            new_adj_list.reserve(_graph_store->get_neighbours((location_t)old).size());
```

Comment:
instead of get_nbrs().size, is it useful to have a num_nbrs method in graph store?

Reply:
I think this should be fine as well because we are returning a const reference so there is no way this is modified.